### PR TITLE
Date and time format configuration

### PR DESCRIPTION
**Pull request by michalmilc90**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | [EZP-30248](https://jira.ez.no/browse/EZP-30248)
| Versions      | master, 2.5

**Sections updated:** Back Office config, Extending eZ Platform UI, Twig Functions Reference
**Update:** added description on how to use other method for configuring date and time format.

### CODE REVIEW

File: docs/guide/config_back_office.md [hunk 1]
```diff
@@ -102,13 +102,91 @@ ezpublish:
...
+- `ez_short_time`
+- `ez_full_datetime`
+- `ez_full_date`
+- `ez_full_time`
+
+The following are examples of using the filters:
+
+``` php hl_lines="3 6"
+<div>
+    // Date formatted in a preferred time zone and short datetime format:
```
Comment (DominikaK) [inline docs/guide/config_back_office.md]:
```suggestion
    // Date formatted in the preferred time zone and short datetime format:
```

File: docs/guide/config_back_office.md [hunk 2]
```diff
@@ -102,13 +102,91 @@ ezpublish:
...
+You can also configure date and time format by using the following services:
+
+- `@ezplatform.user.settings.short_datetime_format.formatter`
+- `@ezplatform.user.settings.short_datet_format.formatter`
+- `@ezplatform.user.settings.short_time_format.formatter`
+- `@ezplatform.user.settings.full_datetime_format.formatter`
+- `@ezplatform.user.settings.full_date_format.formatter`
+- `@ezplatform.user.settings.full_time_format.formatter`
+
+To use them, create a `src/AppBundle\Service\MyService.php` file containing:
```
Comment (DominikaK) [inline docs/guide/config_back_office.md]:
```suggestion
To use them, create an `src\AppBundle\Service\MyService.php` file containing:
```

File: docs/guide/config_back_office.md [hunk 3]
```diff
@@ -102,13 +102,91 @@ ezpublish:
...
-                    date_format: 'dd/mm/yyy'
+                    date_format: 'dd/mm/yyyy'
                     time_format: 'hh:mm'
                 full_datetime_format:
-                    date_format: 'dd/mm/yyy'
+                    date_format: 'dd/mm/yyyy'
                     time_format: 'hh:mm'
 ```
 
+You can use one of the following filters to format date and time:
```
Comment (pawbuj) [inline docs/guide/config_back_office.md]: ..... following Twig filters ....

File: docs/guide/config_back_office.md (hunk 2 above)
Comment (DominikaK) [inline docs/guide/config_back_office.md]: Also, maybe move this to Extending eZ Platform? This is not strictly speaking config

File: docs/guide/config_back_office.md [hunk 4]
```diff
@@ -102,13 +102,91 @@ ezpublish:
...
+    
+    // Date formatted in UTC and preferred short datetime format:
+    {{ content.versionInfo.creationDate|ez_short_datetime('UTC') }}
+</div>
+```
+
+The filters accept an optional `timezone` parameter for displaying date and time in a chosen time zone.
+The default time zone is set in the User settings menu.
+
+You can also configure date and time format by using the following services:
```
Comment (pawbuj) [inline docs/guide/config_back_office.md]: You can also format date and time by using the ...

File: docs/guide/config_back_office.md [hunk 5]
```diff
@@ -102,13 +102,91 @@ ezpublish:
...
                     time_format: 'hh:mm'
                 full_datetime_format:
-                    date_format: 'dd/mm/yyy'
+                    date_format: 'dd/mm/yyyy'
                     time_format: 'hh:mm'
 ```
 
+You can use one of the following filters to format date and time:
+
+- `ez_short_datetime`
```
[the remark this answers is not in the document]
Comment (michalmilc90) [reply docs/guide/config_back_office.md]: Good point!

File: docs/guide/config_back_office.md (hunk 2 above)
Comment (pawbuj) [inline docs/guide/config_back_office.md]: To use them you can, for example, create  ...

File: docs/guide/twig_functions_reference.md [hunk 6]
```diff
@@ -226,17 +232,24 @@ It can be used for example to identify the first image in an article to render i
...
+The formats are defined in [user preferences](config_back_office.md#date-and-time-formats).
 
-It accepts `\DateTimeInterface` as argument.
+| Twig filter | Description |
+|-------------|-------------|
+| `ez_full_datetime` | outputs date and time in full format |
+| `ez_full_date` | outputs date in full format |
+| `ez_full_time` | outputs time in full format |
 
+Thee filters accept `\DateTimeInterface` as argument.
```
Comment (DominikaK) [inline docs/guide/twig_functions_reference.md]:
```suggestion
The filters accept `\DateTimeInterface` as argument.
```

File: docs/guide/twig_functions_reference.md [hunk 7]
```diff
@@ -226,17 +232,24 @@ It can be used for example to identify the first image in an article to render i
...
 If the argument is null, the filter returns the current date and time in the selected format.
 
 For example `{{ contentInfo.publishedDate|ez_full_datetime }}` will return `03 May 2019 23:03`.
 
+Thee filters also accept an optional `timezone` parameter for displaying date and time in a chosen time zone.
```
Comment (DominikaK) [inline docs/guide/twig_functions_reference.md]:
```suggestion
The filters also accept an optional `timezone` parameter for displaying date and time in a chosen time zone.
```

File: docs/guide/extending_ez_platform.md [hunk 8]
```diff
@@ -670,3 +671,85 @@ appbundle.components.my_script_component:
...
+- `ez_short_time`
+- `ez_full_datetime`
+- `ez_full_date`
+- `ez_full_time`
+
+The following are examples of using the filters:
+
+``` php hl_lines="3 6"
+<div>
+    // Date formatted in a preferred time zone and short datetime format:
```
Comment (DominikaK) [inline docs/guide/extending_ez_platform.md]:
```suggestion
    // Date formatted in the preferred time zone and short datetime format:
```

File: docs/guide/extending_ez_platform.md [hunk 9]
```diff
@@ -670,3 +671,85 @@ appbundle.components.my_script_component:
...
+You can also format date and time by using the following services:
+
+- `@ezplatform.user.settings.short_datetime_format.formatter`
+- `@ezplatform.user.settings.short_datet_format.formatter`
+- `@ezplatform.user.settings.short_time_format.formatter`
+- `@ezplatform.user.settings.full_datetime_format.formatter`
+- `@ezplatform.user.settings.full_date_format.formatter`
+- `@ezplatform.user.settings.full_time_format.formatter`
+
+To use them, create a `src/AppBundle\Service\MyService.php` file containing:
```
Comment (DominikaK) [inline docs/guide/extending_ez_platform.md]:
```suggestion
To use them, create an `src\AppBundle\Service\MyService.php` file containing:
```

**Pull request merged**